extensions/omni_vfs/migrate/4_table_fs.sql

```diff
@@ -0,0 +1,217 @@
+create table table_fs_filesystems
+(
+    id   integer primary key generated always as identity,
+    name text not null unique
+);
+
+create table table_fs_files
+(
+    id            bigint primary key generated always as identity,
+    filesystem_id integer              not null references table_fs_filesystems (id),
+    filepath      text                 not null check (filepath != '/'),
+    created_at    timestamp,
+    accessed_at   timestamp,
+    modified_at   timestamp,
+    -- disallow 'dir' entries because they are implicit
+    kind          omni_vfs_types_v1.file_kind not null check (kind != 'dir'),
+    unique (filesystem_id, filepath)
+);
+
+create or replace function table_fs_files_trigger() returns trigger as
+$$
+declare
+    result_name text;
+begin
+    new.filepath := omni_vfs.canonicalize_path(new.filepath, absolute => true);
+    select
+        filepath
+    from
+        omni_vfs.table_fs_files
+    where
+        filesystem_id = new.filesystem_id and
+        (
+            -- file can't be inside a file
+            new.filepath like (filepath || '/%')
+            or
+            -- dir can't be a file
+            filepath like (new.filepath || '/%')
```

Comment:
both are necessary to ensure correctness, but it will become slower as the table grows because `new.filepath like (filepath || '/%')` always leads to a sequential scan

Reply:
can we enforce these rules on inserts/updates to the table?

Reply:
It is enforced on inserts and updates (of some columns) only

Reply:
I apologize, I misread the name of the function above. I see this is the trigger itself. So the problem identified here is on the trigger side making inserts/updates, slow, correct?

That is, it is separate from the performance of listing files. Correct?

Reply:
Yes, this is separate from listing performance